### PR TITLE
Create Reimbursement contract request

### PR DESCRIPTION
**Pull request by PavelVesnin**

### CODE REVIEW

File: apps/core/lib/core/contract_requests/contract_requests.ex [hunk 1]
```diff
@@ -74,12 +46,31 @@ defmodule Core.ContractRequests do
...
 
   def search(search_params) do
-    with %Ecto.Changeset{valid?: true} = changeset <- Search.changeset(search_params),
+    with %Changeset{valid?: true} = changeset <- Search.changeset(search_params),
          %Page{} = paging <- search(changeset, search_params, CapitationContractRequest) do
       {:ok, paging}
     end
   end
 
+  def get_by_id(headers, client_type, id) do
```
Comment (progsmile) [inline apps/core/lib/core/contract_requests/contract_requests.ex]: Suppose we should extract 3 modules: CapitationContractRequest, ReimbursmentContractRequest, ContractRequest - with shared functionality 
It's like going mess here
What do you think?

File: apps/core/lib/core/contract_requests/contract_requests.ex [hunk 2]
```diff
@@ -782,14 +740,38 @@ defmodule Core.ContractRequests do
...
+  end
+
   defp terminate_pending_contracts(params) do
     # TODO: add index here
 
+    schema =
+      case params["type"] do
+        @capitation -> CapitationContractRequest
+        @reimbursement -> ReimbursementContractRequest
+      end
```
[the remark this answers is not in the document]
Comment (PavelVesnin) [reply apps/core/lib/core/contract_requests/contract_requests.ex]: Let it crash. Allowed types validated on request, so unsupported typed shouldn't be here

File: apps/core/lib/core/contract_requests/contract_requests.ex [hunk 3]
```diff
@@ -74,12 +46,31 @@ defmodule Core.ContractRequests do
...
+    client_id = get_client_id(headers)
+
+    with {:ok, %CapitationContractRequest{} = contract_request} <- get_contract_request(client_id, client_type, id) do
+      {:ok, contract_request, preload_references(contract_request)}
+    end
+  end
+
+  def get_by_id(id), do: Repo.get(CapitationContractRequest, id)
+
+  def get_by_id!(id), do: Repo.get!(CapitationContractRequest, id)
```
[the remark this answers is not in the document]
Comment (PavelVesnin) [reply apps/core/lib/core/contract_requests/contract_requests.ex]: I don't like this idea.
`get_by_id` capsulate type of schema. If it takes it from param - it's better to use Repo directly.

File: apps/core/lib/core/contract_requests/contract_requests.ex (hunk 2 above)
[the remark this answers is not in the document]
Comment (progsmile) [reply apps/core/lib/core/contract_requests/contract_requests.ex]: how about schema functions?
```
def contract_module(@type_capitation), do: CapitationContract
def contract_module(@type_reimbursment), do: ReimbursementContract
```
They will be needed in get_by_id

**Pull request merged**